frontend/src/Dashboard/HistoricData.js

```diff
@@ -92,79 +110,100 @@ const formatData = (data) => {
       icon: VscDashboard,
       titleText: 'Szélsebesség',
       text: data.windData.windSpeed,
-      unit: 'km/h"',
+      unit: 'km/h',
     },
-
+  ];
+};
+const soilData = (data) => {
+  return [
     {
       icon: WiThermometer,
       titleText: 'Talaj hőmérséklet 0cm',
-      text: data.soilData.soilTemperature0cm,
+      text: data.soilData.soilTemperature0cm.toFixed(2),
       unit: <>&#8451;</>,
     },
 
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 30cm',
-      text: data.soilData.soilMoisture30cm,
+      text: data.soilData.soilMoisture30cm.toFixed(2),
       unit: 'V/V %',
     },
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 60cm',
-      text: data.soilData.soilMoisture60cm,
+      text: data.soilData.soilMoisture60cm.toFixed(2),
       unit: 'V/V %',
     },
 
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 90cm',
-      text: data.soilData.soilMoisture90cm,
+      text: data.soilData.soilMoisture90cm.toFixed(2),
       unit: 'V/V %',
     },
 
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 120cm',
-      text: data.soilData.soilMoisture120cm,
+      text: data.soilData.soilMoisture120cm.toFixed(2),
       unit: 'V/V %',
     },
-    {
-      icon: HiSun,
-      titleText: 'Besugárzás',
-      text: data.miscData.irradiation,
-      unit: 'W/m2',
-    },
-
-    {
-      icon: HiSun,
-      titleText: 'Fényegység',
-      text: data.miscData.lightUnit,
-      unit: 'cd',
-    },
-
+  ];
+};
+const batteryData = (data) => {
+  return [
     {
       icon: RiBattery2ChargeLine,
       titleText: 'Napelem töltő feszültség',
-      text: data.batteryData.solarCellChargingVoltage,
+      text: data.batteryData.solarCellChargingVoltage.toFixed(2),
       unit: 'V',
     },
 
     {
       icon: GiCarBattery,
       titleText: 'Külső akkufeszültség',
-      text: data.batteryData.externalBatteryVoltage,
+      text: data.batteryData.externalBatteryVoltage.toFixed(2),
       unit: 'V',
     },
 
     {
       icon: GiCarBattery,
       titleText: 'Belső akkufeszültség',
-      text: data.batteryData.internalBatteryVoltage,
+      text: data.batteryData.internalBatteryVoltage.toFixed(2),
       unit: 'V',
     },
   ];
 };
 
+const windDirection = (wind) => {
+  if (wind < 22.5 || wind > 337.5) {
+    return 'Északi';
+  } else if (wind < 67.5) {
+    return 'Észak-Keleti';
+  } else if (wind < 112.5) {
+    return 'Keleti';
+  } else if (wind < 157.5) {
+    return 'Dél-Keleti';
+  } else if (wind < 202.5) {
+    return 'Déli';
+  } else if (wind < 247.5) {
+    return 'Dél-Nyugati';
+  } else if (wind < 292.5) {
+    return 'Nyugati';
+  } else if (wind < 337.5) {
+    return 'Észak-Nyugati';
+  }
+};
+
+const isFreeze = (freeze) => {
```

Comment:
És mi történik az else ágban? Ne hagyjunk method-ot return nélkül. Főleg ne egy típustalan nyelvben, ahol amúgy is bárhol megtörténhet bármi. Kezeljük le hogy ha null érték vagy undifined jön, de soha ne hagyjunk így behalni egy functiont. Egy típusos nyelvben még belefér, hogy tudjuk hogy tuti van érték, és tuti egy szám, de JS-re ez nem igaz. Legalább az else if-et cseréljük le else-re.

Reply:
Javitva, az elején levizsgálva.

Reply:
A gond ugyan az mint a fentebbinel.

Reply:
Javitva